lib/datasets/aozora-bunko.rb

```diff
@@ -0,0 +1,220 @@
+      :person_copyrighted,
+      # 底本名1,底本出版社名1,底本初版発行年1,入力に使用した版1,校正に使用した版1,底本の親本名1,底本の親本出版社名1,底本の親本初版発行年1,
+      :original_book_name1,
+      :original_book_publisher_name1,
+      :original_book_first_published_date1,
+      :used_version_for_registration1,
+      :used_version_for_proofreading1,
+      :base_of_original_book_name1,
+      :base_of_original_book_publisher_name1,
+      :base_of_original_book_first_published_date1,
+      # 底本名2,底本出版社名2,底本初版発行年2,入力に使用した版2,校正に使用した版2,底本の親本名2,底本の親本出版社名2,底本の親本初版発行年2,
+      :original_book_name2,
+      :original_book_publisher_name2,
+      :original_book_first_published_date2,
+      :used_version_for_registration2,
+      :used_version_for_proofreading2,
+      :base_of_original_book_name2,
+      :base_of_original_book_publisher_name2,
+      :base_of_original_book_first_published_date2,
+      # 入力者,校正者,
+      :registered_person_name,
+      :proofreader_name,
+      # テキストファイルURL,テキストファイル最終更新日,テキストファイル符号化方式,テキストファイル文字集合,テキストファイル修正回数,
+      :text_file_url,
+      :last_text_file_updated_date,
+      :text_file_character_encoding,
+      :text_file_character_set,
+      :text_file_updating_count,
+      # XHTML/HTMLファイルURL,XHTML/HTMLファイル最終更新日,XHTML/HTMLファイル符号化方式,XHTML/HTMLファイル文字集合,XHTML/HTMLファイル修正回数
+      :html_file_url,
+      :last_html_file_updated_date,
+      :html_file_character_encoding,
+      :html_file_character_set,
+      :html_file_updating_count
+    )
+
+    class Record
+      def initialize(*args)
+        super
+        @text = nil
+        @html = nil
+      end
+
+      def text
+        return @text unless @text.nil?
+        return @text if text_file_url.nil? || text_file_url.empty?
+
+        # when url is not zip file, it needs to open web page by brower and has to download
+        # e.g. https://mega.nz/file/6tMxgAjZ#PglDDyJL0syRhnULqK0qhTMC7cktsgqwObj5fY_knpE
+        return @text unless text_file_url.end_with?('.zip')
+
+        downloader = Downloader.new(text_file_url)
+        downloader.download(text_file_output_path)
+
+        ZipExtractor.new(text_file_output_path).extract_one_file do |input|
+          @text = input.read.encode(Encoding::UTF_8, normalize_encoding(text_file_character_encoding))
+        end
+
+        @text
+      end
+
+      def html
+        return @html unless @html.nil?
+        return @html if html_file_url.nil? || html_file_url.empty?
+
+        downloader = Downloader.new(html_file_url)
+        downloader.download(html_file_output_path)
+        @html = IO.read(html_file_output_path).encode(Encoding::UTF_8, normalize_encoding(html_file_character_encoding))
+
+        @html
+      end
+
+      def person_copyrighted
+        self[:person_copyrighted] == 'あり'
+      end
+
+      def copyrighted
+        self[:copyrighted] == 'あり'
+      end
+
+      def clear_text_file!
+        cache_path.remove_file(text_file_name)
+      end
+
+      def clear_html_file!
+        cache_path.remove_file(html_file_name)
+      end
```

Comment:
We don't need to provide these specific cache management features.
Providing only `clear_cache!` is enough.

BTW, I rethink about caching. `Datasets::AozoraBunko#clear_cache!` clears all related caches including caches for each `Record` will be enough. We don't need to provide `Record#clear_cache!`. How about the following?

```ruby
class Record
  private attr_writer(:cache_path)

  private
  def cache_base_dir
    @cache_path.base_dir + title_id + person_id
  end

  def text_file_output_path
    cache_base_dir + text_file_name
  end

  def html_file_output_path
    cache_base_dir + html_file_name
  end
end

def each
  # ...
  record = Record.new(*row.fields)
  record.cache_path = cache_path
  yield(record)
  # ...
end
```

Reply:
done